zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/LearnerHandler.java

```diff
@@ -525,15 +525,16 @@ public void run() {
             long newEpoch = learnerMaster.getEpochToPropose(this.getSid(), lastAcceptedEpoch);
             long newLeaderZxid = ZxidUtils.makeZxid(newEpoch, 0);
 
-            if (this.getVersion() < 0x10000) {
-                // we are going to have to extrapolate the epoch information
-                long epoch = ZxidUtils.getEpochFromZxid(zxid);
-                ss = new StateSummary(epoch, zxid);
-                // fake the message
-                learnerMaster.waitForEpochAck(this.getSid(), ss);
+            /**
+             * When the Leader switches to a new mode
+             * lower version Observers or Followers are not allowed to join the cluster, thus the zxid bit length has changed.
+             */
+            if (this.getVersion() < 0x11000) {
+                LOG.error("sid:{} version too old", this.sid);
+                return;
             } else {
                 byte[] ver = new byte[4];
-                ByteBuffer.wrap(ver).putInt(0x10000);
+                ByteBuffer.wrap(ver).putInt(0x11000);
```

Comment:
The patch doesn't alter anything in the protocol, but you still have to increase protocol version number in order to reject old followers. Odd, but seams reasonable.